File: Data Structures in C/Linked Lists/Singly Linked List/SinglyLinkedListDeletion.c
/*
   This program shows how to delete any node of a Singly Linked List data structure in C
   Author - Rajarshi Sengupta
   Github - https://github.com/rajarshisg
   Date - 19/08/2020 (dd/mm/yyyy)
*/
#include<stdio.h>
#include<stdlib.h>
typedef struct Node
{
    int data;
    struct Node *next;
}Node;
Node *delete(Node *head,int pos)
{
    if(head==NULL)
    return NULL;
    Node *n,*p;
    n=head;
    if(pos==1)
    {
      
        head=head->next;
        free(n);
    }
    else
    {
        int k=1;
        while(n!=NULL && k<pos-1)
        {
            n=n->next;
            k++;
        }
        if(n==NULL)
        return head;
        p=n->next;
        n->next=p->next;
        free(p);
    }
    return head;  
}
void traverse(Node *head)
{
     while(head!=NULL)
    {
        printf("%d ",head->data);
        head=head->next;
    }
    printf("\n");
}
int main()
{
    printf("Enter the number of nodes you want :\n");
    int num;
    scanf("%d",&num);
     Node *head=NULL;
     Node *n=NULL;
     Node *m=NULL;
    printf("Enter the elements :\n");
    for(int i=0;i<num;i++)
    {
        if(head==NULL){
            n=(Node *)malloc(sizeof(Node));
            scanf("%d",&n->data);
            n->next=NULL;
            head=n;
        }
        else
        {
           m=(Node *)malloc(sizeof(Node));
           scanf("%d",&m->data);
           m->next=NULL;
           n->next=m;
           n=n->next;
        } 
    }
    printf("The Linked List is :\n");
    traverse(head);
    printf("Enter position you want to delete :\n");
    int pos;
    scanf("%d",&pos);
    head=delete(head,pos);
    printf("After deletion new Linked List is :\n");
    traverse(head);
    return 0;
}

File: Data Structures in JAVA/Linked Lists/Singly Linked Lists/src/SinglyLinkedListDeletion.java
/*
   This program deletes a node at any given position in a Singly Linked List in JAVA
   Author - Rajarshi Sengupta
   Github - https://github.com/rajarshisg
   Date - 20/08/2020 (dd/mm/yyyy)
*/
import java.util.Scanner;
public class SinglyLinkedListDeletion {
	void traverse(Node<Integer> head) {
    	while(head!=null)
    	{
           System.out.print(head.data+" ");
           head=head.next;
    	}
    	System.out.println();
    }
    Node<Integer> delete(Node<Integer> head,int pos)
    {
    	if(head==null)
    		return null;
    	if(pos==1)
    	{
    		Node n=head;
    		head=head.next;
    		n=null;
    	}
    	else
    	{
    		int k=1;
    		Node<Integer> h=head;
    		while(k!=pos-1 && h.next!=null) {
    			h=h.next;
    			k++;
    		}
    		if(h.next!=null) {
    		Node n=h.next;
    		h.next=n.next;
    		n=null;
    		}
    	}
    	return head;
    }
    public static void main(String args[]) {
       Scanner sc=new Scanner(System.in);
 	   Node<Integer> head=null,n=null,m=null;
 	   System.out.println("Enter the number of nodes :");
 	   int num=sc.nextInt();
 	   System.out.println("Enter the elements :");
 	   for(int i=1;i<=num;i++)
 	   {
 		   if(head==null)
 		   {
 			   n=new Node(sc.nextInt());
 			   head=n;
 		   }
 		   else
 		   {
 			   m=new Node(sc.nextInt());
 			   n.next=m;
 			   n=n.next;
 		   }
 	   }
 	   System.out.println("The Linked List is :");
 	   SinglyLinkedListDeletion obj=new SinglyLinkedListDeletion();
 	   obj.traverse(head);
 	   System.out.println("Enter position you want to delete :");
 	   int pos=sc.nextInt();
 	   head=obj.delete(head,pos);
 	   System.out.println("The Linked List is :");
	   obj.traverse(head);
    }
}
